extensions/interactions/MultipleChoiceInput/protractor.js

```diff
@@ -36,14 +36,19 @@ var customizeInteraction = function(elem, richTextInstructionsArray) {
 // the options.
 var expectInteractionDetailsToMatch = function(
     elem, richTextInstructionsArray) {
-  elem.all(by.repeater('choice in $ctrl.choices track by $index'))
+  elem.all(by.repeater('choice in $ctrl.choices'))
     .then(function(optionElements) {
       expect(optionElements.length).toEqual(richTextInstructionsArray.length);
+      var promises = [];
       for (var i = 0; i < optionElements.length; i++) {
-        forms.expectRichText(optionElements[i].element(by.css(
-          '.protractor-test-multiple-choice-option'
-        ))).toMatch(richTextInstructionsArray[i]);
+        promises.push(optionElements[i].element(by.css(
+          '.protractor-test-multiple-choice-option')).getText());
       }
+      protractor.promise.all(promises).then(function(results) {
+        for (var j = 0; j < promises.length; j++) {
```

Comment:
Can you just sort the two arrays and check that they are equal? More direct, and prevents some corner cases from not being handled correctly (like when an item is duplicated).

Reply:
Done.